Sources/AsyncKit/ConnectionPool/EventLoopConnectionPool.swift

```diff
@@ -242,40 +255,37 @@ public final class EventLoopConnectionPool<Source> where Source: ConnectionPoolS
     ///     - connection: Connection to release back to the pool.
     ///     - logger: For trace and debug logs.
     public func releaseConnection(_ connection: Source.Connection, logger: Logger) {
-        // dispatch to event loop thread if necessary
-        guard self.eventLoop.inEventLoop else {
-            return self.eventLoop.execute {
-                self.releaseConnection(connection, logger: logger)
-            }
+        if self.eventLoop.inEventLoop {
+            self._releaseConnection0(connection, logger: logger)
+        } else {
+            self.eventLoop.execute { self._releaseConnection0(connection, logger: logger) }
         }
+    }
+    
+    private func _releaseConnection0(_ connection: Source.Connection, logger: Logger) {
+        self.eventLoop.assertInEventLoop()
         
-        // synchronize access to available / active connection checks
+        // If the pool has shut down, just close the connection and return
         guard !self.didShutdown else {
-            // this pool is closed and we are responsible for closing all
-            // of our connections
-            _ = connection.close()
+            if !connection.isClosed {
+                _ = connection.close()
```

Comment:
Do we care about the case where closing the connection fails?

Reply:
I don't think it matters if we're shutting down the entire pool anyhow; I'm not even sure what error _can_ occur during closing that we'd care about (or really at all) 🤷‍♀️